src/main/java/fi/aalto/cs/apluscourses/utils/PluginResourceBundle.java

```diff
@@ -11,11 +19,46 @@ private PluginResourceBundle() {
 
   private static final ResourceBundle bundle = ResourceBundle.getBundle("resources");
 
+  private static final Map<Project, ResourceBundle> bundles = new HashMap<>();
+
+  public static final String CUSTOM_RESOURCES_FILENAME = "customResources.properties";
+
   public static String getText(@NotNull String key) {
     return bundle.getString(key);
   }
 
+  /**
+   * Gets the text from the project's custom resource bundle if it exists,
+   * else gets the default text.
+   */
+  public static String getText(@NotNull String key, @NotNull Project project) {
+    var customText = Optional.ofNullable(bundles.get(project))
+        .map(customBundle -> customBundle.getString(key));
+    return customText.isEmpty() ? bundle.getString(key) : customText.get();
```

Comment:
Nit: I'd consider it better design if one of these overloads referred to the other. So, instead of `bundle.getString(key)` here, `getText()key` was called, or, perhaps even preferably, `project` in the overload allowed nulls (i.e. no customization), and the other overload just called `getText(key, null)`.

The same goes with `getAndReplaceText`.